src/track/taglib/trackmetadata_common.h

```diff
@@ -0,0 +1,202 @@
+#pragma once
+
+#include <taglib/tag.h>
+#include <taglib/tstringlist.h>
+
+#include <QByteArray>
+#include <QImage>
+#include <QString>
+
+#if defined(__EXTRA_METADATA__)
+// UUID -> QString
+#include "util/compatibility.h"
+#endif // __EXTRA_METADATA__
+
+// TagLib has support for the Ogg Opus file format since version 1.9
+#define TAGLIB_HAS_OPUSFILE \
+    ((TAGLIB_MAJOR_VERSION > 1) || ((TAGLIB_MAJOR_VERSION == 1) && (TAGLIB_MINOR_VERSION >= 9)))
+
+// TagLib has support for hasID3v2Tag()/ID3v2Tag() for WAV files since version 1.9
+#define TAGLIB_HAS_WAV_ID3V2TAG \
+    (TAGLIB_MAJOR_VERSION > 1) || ((TAGLIB_MAJOR_VERSION == 1) && (TAGLIB_MINOR_VERSION >= 9))
+
+#include "track/trackmetadata.h"
+
+namespace mixxx {
+
+namespace taglib {
+
+inline QString toQString(
+        const TagLib::String& tString) {
+    if (tString.isNull()) {
+        // null -> null
+        return QString();
+    } else {
+        return TStringToQString(tString);
+    }
+}
+
+TagLib::String toTString(
+        const QString& str);
+
+// Returns the first element of TagLib string list that is not empty.
```

Comment:
```suggestion
/// Returns the first element of TagLib string list that is not empty.
```

... and for all other comments that document classes/functions/enums/global constants.